### PR TITLE
new resources: keycloak_identity_provider, keycloak_identity_provider_mapper, added username/password auth

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**

### CODE REVIEW

**Comment by mrparkers** [comment]
Hi @AndrewChubatiuk, thanks for the PR!

I have to admit that I am not very familiar with this feature of Keycloak, so I will try to spend some time getting up to speed in order to leave you a proper review.  I'll probably leave a few comments here and there throughout the next few days.

File: example/main.tf [hunk 1]
```diff
@@ -4,6 +4,16 @@ provider "keycloak" {
 	url           = "http://localhost:8080"
 }
 
+//Or you can use basic auth instead
```
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply example/main.tf]: Don't really want to import json after keycloak setup in ECS, using basic auth allows to setup all resources with terraform

File: example/main.tf [hunk 2]
```diff
@@ -4,6 +4,16 @@ provider "keycloak" {
...
+
+//provider keycloak {
+//  client_id = "admin-cli"
+//  username  = "<your_keycloak_username>"
+//  password  = "<your_keycloak_password>"
+//  url   = "https://keycloak.host"
+//  realm = "master"
```
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply example/main.tf]: https://github.com/mrparkers/terraform-provider-keycloak/pull/64

**Comment by mrparkers** [comment]
I will give this a thorough review after #64 is merged

**Comment by tomrutsaert** [comment]
Any news on the progress of this pull request?

**Comment by tomrutsaert** [comment]
Yes, I am looking into to using your project, and of the things I would need is this and an the smtp settings(other pull request),
There are other things, but I would add those myself and make pull requests for them.

**Comment by mrparkers** [comment]
Is there a particular identity provider you're looking to use?  The only reason I'm asking is because I am unsure whether or not it would make sense to split identity providers into multiple resources (ex: `keycloak_identity_provider_facebook` and `keycloak_identity_provider_google`.  Most of the API fields are similar, but there are a few key differences between some of them, and generally I favor multiple less-complicated resources over a single resource that tries to handle every use case.

I'll look into it some more, but I'd be happy to write an initial PoC using whatever identity providers you happen to need support for.

**Comment by tomrutsaert** [comment]
I agree, 
In the future we will need multiple identity providers, but at first we were planning to start with google.

**Comment by mrparkers** [comment]
Perfect.  I'll try and find some time this week to work on it.  If you don't want to wait, feel free to take a stab at it and submit a PR for `keycloak_identity_provider_google` (or `keycloak_google_identity_provider`, not sure which sounds better).

As for the realm SMTP settings, I can try and wrap that up this week too.

**Pull request closed without merging**